instant-xml/src/lib.rs

```diff
@@ -181,8 +200,163 @@ pub struct FieldContext<'xml> {
     pub attribute: Option<FieldAttribute<'xml>>,
 }
 
+pub enum EntityType {
+    Element,
+    Attribute,
+}
+
+pub enum XMLTagName<'xml> {
+    FieldName,
+    Custom(&'xml str),
+}
+
 pub trait FromXml<'xml>: Sized {
-    fn from_xml(input: &str) -> Result<Self, Error>;
+    const TAG_NAME: XMLTagName<'xml>;
+
+    fn from_xml(_input: &str) -> Result<Self> {
+        unimplemented!();
+    }
+
+    fn deserialize(deserializer: &mut Deserializer, kind: EntityType) -> Result<Self>;
```

Comment:
Is there a reason `kind` cannot be part of the `Deserializer`?

Reply:
Mostly because I wanted to avoid state-like processing, please take a look at this commit: https://github.com/InstantDomain/instant-xml/pull/7/commits/266443baeff1b2a61a7007bf5312b6b6622e1eda
Maybe you have some better idea of how to approch this?

Reply:
done